src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -249,4 +253,125 @@ public override int GetHashCode ()
     /// Gets the type metadata cache for the runtime.
     /// </summary>
     public static ITypeMetadataCache Cache => cache;
```

Comment:
This is more of a lookup than a cache, as it doesn't invalidate. Consider changing the name to avoid misleading interpretation.

Reply:
It's technically Memoization which is a form of caching, but is it really that confusing?